### ISSUE TITLE
UniswapLP - failed to remove liquidity (One of the tokens in this position isn't recognized)

### ISSUE THREAD

**Issue by nikspz** [hummingbot/gateway#154]
**Describe the bug**
UniswapLP - failed to remove liquidity

**Steps To Reproduce**
0. You Have added liquidity with id 123456
1. Clone and install latest development Client + gateway
2. set it up gateway to ONLINE with generate certs and yarn start
3. connect uniswap arbitrum_one
4. set up GATEWAY_CERT, GATEWAY_KEY and ETH_ADDRESS
5. change eth_uniswap_position network to arbitrum_one and tokenId to 123456
6. execute `curl -s -X POST -k --key $GATEWAY_KEY --cert $GATEWAY_CERT -H "Content-Type: application/json" -d "$(envsubst < ./requests/eth_uniswap_position.json)" https://localhost:15888/amm/liquidity/remove | jq`

Actual:
Failed to remove liquidity with error:

```
"httpErrorCode": 503,
  "errorCode": 1099,
  "stack": "Error: One of the tokens in this position isn't recognized. $token0: undefined, $token1: undefined\n    at UniswapLP.<anonymous> (/home/nikita/gwtest1/dist/src/connectors/uniswap/uniswap.lp.helper.js:285:23)\n    at Generator.next (<anonymous>)\n    at fulfilled (/home/nikita/gwtest1/dist/src/connectors/uniswap/uniswap.lp.helper.js:28:58)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
```

![image](https://github.com/hummingbot/gateway/assets/83953535/f424617a-51b1-44ae-a008-494cd4f21a24)
**Release version**
dev-1.18.0

**Comment by fengtality** [comment]
@nikspz Can you provide more info about the position you added and tried to remove?

- transaction hash for the add position tx?
- more info about the position created (tokens, address, etc)

**Comment by nikspz** [comment]
@fengtality position added: 
ID: 719721
Txhash: https://arbiscan.io/tx/0xbdb5878f9c30c9229783296e7241023e59fa27459e47635582398bdb3840686b


[logs_gateway_app.log](https://github.com/hummingbot/gateway/files/12028824/logs_gateway_app.log)
![image](https://github.com/hummingbot/gateway/assets/83953535/5e3df48f-4100-459d-a7e1-606e142a602d)

**Comment by nikspz** [comment]
tokens: 
ARB: 
{
         "chainId":42161,
         "address":"0x912ce59144191c1204e64559fe8253a0e49e6548",
         "name":"Arbitrum",
         "symbol":"ARB",
         "decimals":18,
         "logoURI":"https://assets.coingecko.com/coins/images/16547/thumb/photo_2023-03-29_21.47.00.jpeg?1680097630"
      }

WETH:

{
         "chainId":42161,
         "address":"0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
         "name":"WETH",
         "symbol":"WETH",
         "decimals":18,
         "logoURI":"https://assets.coingecko.com/coins/images/2518/thumb/weth.png?1628852295"
      },

**Comment by nikspz** [comment]
get position request:
```
curl -s -X POST -k --key $GATEWAY_KEY --cert $GATEWAY_CERT -H "Content-Type: application/json" -d "$(envsubst < ./requests/eth_uniswap_position.json)" https://localhost:15888/amm/liquidity/position | jq

```
![image](https://github.com/hummingbot/gateway/assets/83953535/fd1eb02b-71bf-4c9f-a214-86c35ba9b409)


get pool price: 

```

curl -s -X POST -k --key $GATEWAY_KEY --cert $GATEWAY_CERT -H "Content-Type: application/json" -d "$(envsubst < ./requests/eth_uniswap_pool_price.json)" https://localhost:15888/amm/liquidity/price | jq

```

0x82aF49447D8a07e3bd95BD0d56f35241523fBab1 (https://arbiscan.io/address/0x82aF49447D8a07e3bd95BD0d56f35241523fBab1)
0x912CE59144191C1204E64559FE8253a0e49E6548 (https://arbiscan.io/address/0x912CE59144191C1204E64559FE8253a0e49E6548)

![image](https://github.com/hummingbot/gateway/assets/83953535/aa448b57-b788-4d8c-aeec-474bbfdcd972)

uniswapLP Client strategy status

![image](https://github.com/hummingbot/gateway/assets/83953535/ecb9f61b-2b34-4a61-9e00-32bf516ad82d)

**Comment by fengtality** [comment]
@vic-en Can you take this bounty? I think this issue may be related to the chain configuration change, since the call to `getTokenByAddress` now returns undefined in https://github.com/hummingbot/gateway/blob/main/src/connectors/uniswap/uniswap.lp.ts#L74.

**Comment by ZAABY** [comment]
> ## Bounty
> * Description: Fix the UniswapLP connector code so that removing liquidity works
> * Sponsor: Hummingbot Foundation
> * Bounty Rationale: Uniswap is the Gold DEX connector for Epoch 5
> * Bounty Amount: 50,000 HBOT (P2)
> * Developer Payment: 50,000 HBOT (100%)
> * Status: Open
> 
> @vic-en Can you take this bounty? I think this issue may be related to the chain configuration change, since the call to `getTokenByAddress` now returns undefined in https://github.com/hummingbot/gateway/blob/main/src/connectors/uniswap/uniswap.lp.ts#L74.

Bounty Accepted!

**Comment by ZAABY** [comment]
So, from my understanding till now I have traced back the error to get contact function. I think this is were the problem is. So, I am giving a try to get gateway interact directly with the pool instead of via router (Which it's doing rn!)

**Comment by ZAABY** [comment]
Hello Everyone!
So I came to some solution but facing the attached error. Can any one help me what to do to get out of it?
![WhatsApp Image 2023-07-20 at 01 51 50](https://github.com/hummingbot/gateway/assets/96419757/9170c1d4-7846-453d-9032-f41d1a994bf6)

**Comment by fengtality** [comment]
yes it looks like a JS scoping issue as this traces back to the
this.storedTokenList call

the task is to identity root cause and fix this issue.

if you don't have the JS experience to solve this, let me know and we will
reassign to a dev with more experience with Gateway

**Comment by ZAABY** [comment]
I resign!

**Comment by nikspz** [comment]
@ZAABY unassigned from this bounty

**Comment by nikspz** [comment]
Assigned to @vic-en , please be informed:

- Since you’re assigned to this bounty exclusively, please keep us informed on your progress over the next few days
- Let us know if you don’t have time, and we will unassigned and keep it open for other devs
- Only one dev can be assigned to one bounty

Thanks for your contribution!

**Comment by nikspz** [comment]
fix #169 was merged to the latest development branch

**Comment by nikspz** [comment]
Thanks for your participation, the bounty has been sent to @vic-en. The fix has been deployed on development branch and next hummingbot version 1.19.0
